src/main/kotlin/uk/gov/justice/digital/hmpps/deliusapi/service/court/report/CourtReportValidationService.kt

```diff
@@ -0,0 +1,30 @@
+package uk.gov.justice.digital.hmpps.deliusapi.service.court.report
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.deliusapi.entity.Offender
+import uk.gov.justice.digital.hmpps.deliusapi.exception.validation.CourtReportValidationException
+import uk.gov.justice.digital.hmpps.deliusapi.exception.validation.ValidationErrorCodes
+import uk.gov.justice.digital.hmpps.deliusapi.repository.OffenderRepository
+import uk.gov.justice.digital.hmpps.deliusapi.repository.court.report.CourtReportRepository
+import kotlin.jvm.Throws
+
+@Service
+class CourtReportValidationService(
+  val offenderRepository: OffenderRepository,
+  val courtRepository: CourtReportRepository,
+) {
+
+  @Throws(CourtReportValidationException::class)
+  fun validate(crn: String, courtReportId: Long): Offender {
+    // Verify an Offender is available.
+
+    val offender = offenderRepository.findByCrn(crn) ?: throw CourtReportValidationException("Offender with crn $crn cannot be found", ValidationErrorCodes.CRN_NOT_FOUND)
+    val courtReport = courtRepository.getById(courtReportId) ?: throw CourtReportValidationException("Court report with id $courtReportId cannot be found", ValidationErrorCodes.COURT_REPORT_NOT_FOUND)
+    offender.let {
+      if (it.id != courtReport.offenderId) {
+        throw CourtReportValidationException("The courtReport belongs to a different crn than the one entered", ValidationErrorCodes.CRN_NOT_MATCH_OFFENDER_ID)
+      }
+    }
```

Comment:
Small thing, but why wouldn't we do this:
```suggestion
    if (offender.id != courtReport.offenderId) {
      throw CourtReportValidationException("The courtReport belongs to a different crn than the one entered", ValidationErrorCodes.CRN_NOT_MATCH_OFFENDER_ID)
    }
```